<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Graded Assessment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:

[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-\_-Developer_Ed%2BTech-\_-WW_WW-\_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. 

Use the links below to read the data files using the Pandas library. 

* Chicago Census Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Public Schools

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

* Chicago Crime Data

https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01

**NOTE:** Ensure you use the datasets available on the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


Execute the below code cell to avoid prettytable default error.


In [1]:
!pip install ipython-sql prettytable

import prettytable

prettytable.DEFAULT = 'DEFAULT'

### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**


Load the `pandas` and `sqlite3` libraries and establish a connection to `FinalDB.db`


In [9]:
import pandas as pd
import sqlite3
import csv

con = sqlite3.connect("FinalDB.db")
cur = con.cursor()

Load the SQL magic module


In [10]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


Use `Pandas` to load the data available in the links above to dataframes. Use these dataframes to load data on to the database `FinalDB.db` as required tables.


In [12]:
# Links
url_census = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv"
url_schools = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv"
url_crime = "https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv"

# Use Pandas to load the data available in the links above to dataframes = Load CSVs into DataFrames
df_census = pd.read_csv(url_census)
df_schools = pd.read_csv(url_schools)
df_crime = pd.read_csv(url_crime)

# Save DataFrames as tables in the database =  load data on to the database FinalDB.db
df_census.to_sql('CensusData', con, if_exists='replace', index=False)
df_schools.to_sql('PublicSchools', con, if_exists='replace', index=False)
df_crime.to_sql('CrimeData', con, if_exists='replace', index=False)

533

Establish a connection between SQL magic module and the database `FinalDB.db`


In [13]:
%sql sqlite:///FinalDB.db

You can now proceed to the the following questions. Please note that a graded assignment will follow this lab and there will be a question on each of the problems stated below. It can be from the answer you received or the code you write for this problem. Therefore, please keep a note of both your codes as well as the response you generate.


## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [25]:
%sql SELECT COUNT(*) AS Total_Crimes FROM CrimeData;

 * sqlite:///FinalDB.db
Done.


[(533,)]

### Problem 2

##### List community area names and numbers with per capita income less than 11000.


In [28]:
%sql SELECT COMMUNITY_AREA_NAME, COMMUNITY_AREA_NUMBER, PER_CAPITA_INCOME FROM CensusData WHERE PER_CAPITA_INCOME < 11000 ORDER BY PER_CAPITA_INCOME;

 * sqlite:///FinalDB.db
Done.


[('Riverdale', 54.0, 8201),
 ('South Lawndale', 30.0, 10402),
 ('Fuller Park', 37.0, 10432),
 ('West Garfield Park', 26.0, 10934)]

### Problem 3

##### List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis) 


In [87]:
# No problem 3
%sql SELECT CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION FROM CrimeData ORDER BY PRIMARY_TYPE

 * sqlite:///FinalDB.db
Done.


[('HX343113', 'ARSON', 'AGGRAVATED'),
 ('HM195342', 'ARSON', 'BY FIRE'),
 ('HN182437', 'ASSAULT', 'SIMPLE'),
 ('HW493063', 'ASSAULT', 'AGGRAVATED: OTHER DANG WEAPON'),
 ('HJ637197', 'ASSAULT', 'SIMPLE'),
 ('HH829588', 'ASSAULT', 'AGGRAVATED: HANDGUN'),
 ('HZ322965', 'ASSAULT', 'AGGRAVATED: OTHER DANG WEAPON'),
 ('HW415978', 'ASSAULT', 'AGGRAVATED: HANDGUN'),
 ('HL315870', 'ASSAULT', 'AGGRAVATED: OTHER DANG WEAPON'),
 ('HJ222529', 'ASSAULT', 'AGGRAVATED PO: OTHER DANG WEAP'),
 ('HM535930', 'ASSAULT', 'SIMPLE'),
 ('HZ242203', 'ASSAULT', 'AGGRAVATED:KNIFE/CUTTING INSTR'),
 ('HN213069', 'ASSAULT', 'SIMPLE'),
 ('HH343366', 'ASSAULT', 'SIMPLE'),
 ('JA269772', 'ASSAULT', 'AGGRAVATED: HANDGUN'),
 ('HH106347', 'ASSAULT', 'SIMPLE'),
 ('HM507529', 'ASSAULT', 'SIMPLE'),
 ('HP258815', 'ASSAULT', 'SIMPLE'),
 ('HL111533', 'ASSAULT', 'SIMPLE'),
 ('HS479338', 'ASSAULT', 'SIMPLE'),
 ('HZ403432', 'ASSAULT', 'AGGRAVATED: HANDGUN'),
 ('HV462975', 'ASSAULT', 'SIMPLE'),
 ('HW414519', 'ASSAULT', 'SIMPLE'),
 ('HL414306', 'ASSAULT', 'AGGRAVATED: HANDGUN'),
 ('HT315369', 'ASSAULT', 'PRO EMP HANDS NO/MIN INJURY'),
 ('HL330223', 'ASSAULT', 'SIMPLE'),
 ('HK751265', 'ASSAULT', 'AGGRAVATED: OTHER DANG WEAPON'),
 ('HT540548', 'ASSAULT', 'SIMPLE'),
 ('HX398190', 'ASSAULT', 'SIMPLE'),
 ('HK348713', 'ASSAULT', 'SIMPLE'),
 ('HR206683', 'ASSAULT', 'SIMPLE'),
 ('HH154448', 'ASSAULT', 'SIMPLE'),
 ('JA462321', 'ASSAULT', 'AGGRAVATED:KNIFE/CUTTING INSTR'),
 ('HV233824', 'ASSAULT', 'SIMPLE'),
 ('HS451246', 'BATTERY', 'SIMPLE'),
 ('HH316735', 'BATTERY', 'SIMPLE'),
 ('G597730', 'BATTERY', 'SIMPLE'),
 ('G736811', 'BATTERY', 'SIMPLE'),
 ('HT651635', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HZ432273', 'BATTERY', 'SIMPLE'),
 ('G517474', 'BATTERY', 'SIMPLE'),
 ('HP746347', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('G056330', 'BATTERY', 'SIMPLE'),
 ('HN772519', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HZ309936', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HY368284', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HL353697', 'BATTERY', 'SIMPLE'),
 ('HT375934', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('JA188596', 'BATTERY', 'AGG PO HANDS NO/MIN INJURY'),
 ('HL725506', 'BATTERY', 'PRO EMP HANDS NO/MIN INJURY'),
 ('HV395700', 'BATTERY', 'SIMPLE'),
 ('HH696901', 'BATTERY', 'SIMPLE'),
 ('HT390647', 'BATTERY', 'SIMPLE'),
 ('HK815196', 'BATTERY', 'AGGRAVATED:KNIFE/CUTTING INSTR'),
 ('JA490957', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HK103473', 'BATTERY', 'SIMPLE'),
 ('HM556695', 'BATTERY', 'AGGRAVATED: OTHER DANG WEAPON'),
 ('HH558381', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HK498590', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HX455839', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HR703960', 'BATTERY', 'AGGRAVATED:KNIFE/CUTTING INSTR'),
 ('HL646682', 'BATTERY', 'SIMPLE'),
 ('HV445580', 'BATTERY', 'SIMPLE'),
 ('G688026', 'BATTERY', 'AGGRAVATED: OTHER DANG WEAPON'),
 ('HJ466766', 'BATTERY', 'SIMPLE'),
 ('HL632721', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HP579910', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HL718742', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HP714065', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HP716225', 'BATTERY', 'SIMPLE'),
 ('HW481295', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HS568476', 'BATTERY', 'AGGRAVATED: OTHER DANG WEAPON'),
 ('JB213238', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HS475732', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HS595785', 'BATTERY', 'SIMPLE'),
 ('HR408597', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HT542713', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HL240985', 'BATTERY', 'SIMPLE'),
 ('HJ254782', 'BATTERY', 'AGGRAVATED: OTHER DANG WEAPON'),
 ('HK561262', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HW213573', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('G419948', 'BATTERY', 'SIMPLE'),
 ('HL177008', 'BATTERY', 'SIMPLE'),
 ('HJ639869', 'BATTERY', 'SIMPLE'),
 ('HM217017', 'BATTERY', 'SIMPLE'),
 ('HL388623', 'BATTERY', 'SIMPLE'),
 ('HY189363', 'BATTERY', 'SIMPLE'),
 ('HS498285', 'BATTERY', 'DOMESTIC BATTERY SIMPLE'),
 ('HN224900', 'BATTERY', 'AGGRAV

In [40]:
#List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis)
%sql SELECT CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION FROM CrimeData WHERE PRIMARY_TYPE = "OFFENSE INVOLVING CHILDREN";

 * sqlite:///FinalDB.db
Done.


[('HN567387', 'OFFENSE INVOLVING CHILDREN', 'AGG SEX ASSLT OF CHILD FAM MBR'),
 ('HR391350', 'OFFENSE INVOLVING CHILDREN', 'SEX ASSLT OF CHILD BY FAM MBR'),
 ('HM768251', 'OFFENSE INVOLVING CHILDREN', 'CRIM SEX ABUSE BY FAM MEMBER'),
 ('HT394616', 'OFFENSE INVOLVING CHILDREN', 'AGG CRIM SEX ABUSE FAM MEMBER')]

### Problem 4

##### List all kidnapping crimes involving a child?


In [42]:
%sql SELECT CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION FROM CrimeData WHERE PRIMARY_TYPE = "KIDNAPPING";

 * sqlite:///FinalDB.db
Done.


[('HN144152', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER')]

### Problem 5

##### List the kind of crimes that were recorded at schools. (No repetitions)


In [65]:
#No problem 5
%sql SELECT * from CrimeData WHERE LOCATION_DESCRIPTION LIKE '%school%';

 * sqlite:///FinalDB.db
Done.


[(4006321, 'HL353697', '2005-05-04', '077XX S BURNHAM AVE', '460', 'BATTERY', 'SIMPLE', 'SCHOOL, PUBLIC, GROUNDS', 0, 0, 421, 4, 7.0, 43.0, '08B', 1196066.0, 1854156.0, 2005, 41.75469107, -87.55703869, '(41.754691074, -87.557038686)'),
 (4430638, 'HL725506', '2005-11-09', '048XX N FRANCISCO AVE', '484', 'BATTERY', 'PRO EMP HANDS NO/MIN INJURY', 'SCHOOL, PUBLIC, BUILDING', 1, 0, 2031, 20, 40.0, 4.0, '08B', 1156348.0, 1932071.0, 2005, 41.96938944, -87.70048881, '(41.96938944, -87.700488807)'),
 (6644618, 'HP716225', '2008-12-04', '030XX S DR MARTIN LUTHER KING JR DR', '460', 'BATTERY', 'SIMPLE', 'SCHOOL, PUBLIC, BUILDING', 0, 0, 2112, 1, 2.0, 35.0, '08B', 1179306.0, 1885032.0, 2008, 41.83981621, -87.61751617, '(41.839816207, -87.617516172)'),
 (2341955, 'HH639427', '2002-09-10', '005XX N WALLER AVE', '460', 'BATTERY', 'SIMPLE', 'SCHOOL, PUBLIC, BUILDING', 0, 0, 1512, 15, 29.0, 25.0, '08B', 1138238.0, 1903181.0, 2002, 41.89045993, -87.76778089, '(41.890459933, -87.767780886)'),
 (11110571, 'JA460432', '2017-10-05', '076XX S HOMAN AVE', '460', 'BATTERY', 'SIMPLE', 'SCHOOL, PUBLIC, GROUNDS', 0, 0, 835, 8, 18.0, 70.0, '08B', 1155024.0, 1853614.0, 2017, 41.75412154, -87.70746025, '(41.754121535, -87.707460248)'),
 (7399281, 'HS200939', '2010-03-10', '053XX W CONGRESS PKWY', '1320', 'CRIMINAL DAMAGE', 'TO VEHICLE', 'SCHOOL, PUBLIC, GROUNDS', 0, 0, 1522, 15, 29.0, 25.0, '14', 1140822.0, 1897164.0, 2010, 41.8739014, -87.7584391, '(41.873901397, -87.758439102)'),
 (3530721, 'HK577020', '2004-08-23', '016XX W JONQUIL TER', '2024', 'NARCOTICS', 'POSS: HEROIN(WHITE)', 'SCHOOL, PUBLIC, GROUNDS', 1, 0, 2422, 24, 49.0, 1.0, '18', 1163893.0, 1951001.0, 2004, 42.0211776, -87.67220843, '(42.021177601, -87.67220843)'),
 (7502426, 'HS305355', '2010-05-13', '035XX S WASHTENAW AVE', '1821', 'NARCOTICS', 'MANU/DEL:CANNABIS 10GM OR LESS', 'SCHOOL, PUBLIC, BUILDING', 1, 0, 913, 9, 12.0, 58.0, '18', 1158944.0, 1880896.0, 2010, 41.82890791, -87.69234919, '(41.828907913, -87.692349187)'),
 (8082600, 'HT315369', '2011-05-26', '032XX W ADAMS ST', '545', 'ASSAULT', 'PRO EMP HANDS NO/MIN INJURY', 'SCHOOL, PUBLIC, GROUNDS', 0, 0, 1124, 11, 28.0, 27.0, '08A', 1154752.0, 1898890.0, 2011, 41.87837031, -87.70724814, '(41.878370307, -87.707248137)'),
 (7174283, 'HR585012', '2009-10-13', '043XX W 79TH ST', '1330', 'CRIMINAL TRESPASS', 'TO LAND', 'SCHOOL, PUBLIC, GROUNDS', 1, 0, 834, 8, 13.0, 70.0, '26', 1148764.0, 1851854.0, 2009, 41.74941446, -87.7304466, '(41.749414464, -87.730446597)'),
 (2075734, 'HH292682', '2002-04-05', '016XX W BRYN MAWR AV', '2850', 'PUBLIC PEACE VIOLATION', 'BOMB THREAT', 'SCHOOL, PRIVATE, BUILDING', 0, 0, 2012, 20, None, None, '26', None, None, 2002, None, None, None),
 (1806896, 'G635735', '2001-10-22', '0000X N HERMITAGE AV', '2850', 'PUBLIC PEACE VIOLATION', 'BOMB THREAT', 'SCHOOL, PUBLIC, BUILDING', 1, 0, 1333, 12, None, None, '26', 1164687.0, 1900646.0, 2001, 41.88298425, -87.67071922, '(41.882984246, -87.670719218)')]

In [61]:
# List the kind of crimes that were recorded at schools. (No repetitions)

%sql SELECT DISTINCT(PRIMARY_TYPE) from CrimeData WHERE LOCATION_DESCRIPTION LIKE "%school%";


 * sqlite:///FinalDB.db
Done.


[('BATTERY',),
 ('CRIMINAL DAMAGE',),
 ('NARCOTICS',),
 ('ASSAULT',),
 ('CRIMINAL TRESPASS',),
 ('PUBLIC PEACE VIOLATION',)]

### Problem 6

##### List the type of schools along with the average safety score for each type.


In [66]:
#NO PROBLEM 6 - SEE PROBLEM 6 BELOW
%sql SELECT * from PublicSchools  LIMIT 3;

 * sqlite:///FinalDB.db
Done.


[(610038, 'Abraham Lincoln Elementary School', 'ES', '615 W Kemper Pl', 'Chicago', 'IL', 60614, '(773) 534-5720', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610038.pdf', 'Fullerton Elementary Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 1', 'Yes', 'Very Strong', 99.0, 'Very Strong', '99', 'Strong', 74.0, 'Strong', 66.0, 'Weak', '65', 'Strong', '70', 'Strong', '56', 'Average', '47', '96.00%', 2.0, '96.40%', '95.80%', '80.1', '43.3', '89.6', '84.9', '60.7', '62.6', '81.9', '85.2', '52', '62.4', '66.3', '77.9', 69.7, 64.4, 0.2, 0.9, 'Yellow', 'Green', '67.1', '54.5', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 813, 33, 'NDA', 1171699.458, 1915829.428, 41.92449696, -87.64452163, 7, 'LINCOLN PARK', 43, 18, '(41.92449696, -87.64452163)'),
 (610281, 'Adam Clayton Powell Paideia Community Academy Elementary School', 'ES', '7511 S South Shore Dr', 'Chicago', 'IL', 60649, '(773) 535-6650', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610281.pdf', 'Skyway Elementary Network', 'SOUTH SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 54.0, 'Strong', '66', 'Strong', 74.0, 'Very Strong', 84.0, 'Weak', '63', 'Strong', '76', 'Weak', '46', 'Average', '50', '95.60%', 15.7, '95.30%', '100.00%', '62.4', '51.7', '21.9', '15.1', '29', '42.8', '38.5', '27.4', '44.8', '42.7', '14.1', '34.4', 16.8, 16.5, 0.7, 1.4, 'Green', 'Green', '17.2', '27.3', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 521, 46, 'NDA', 1196129.985, 1856209.466, 41.76032435, -87.55673627, 43, 'SOUTH SHORE', 7, 4, '(41.76032435, -87.55673627)'),
 (610185, 'Adlai E Stevenson Elementary School', 'ES', '8010 S Kostner Ave', 'Chicago', 'IL', 60652, '(773) 535-2280', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610185.pdf', 'Midway Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 2', 'No', 'Strong', 61.0, 'NDA', 'NDA', 'Average', 50.0, 'Weak', 36.0, 'Weak', 'NDA', 'NDA', 'NDA', 'Average', '47', 'Weak', '41', '95.70%', 2.3, '94.70%', '98.30%', '53.7', '26.6', '38.3', '34.7', '43.7', '57.3', '48.8', '39.2', '46.8', '44', '7.5', '21.9', 18.3, 15.5, -0.9, -1.0, 'Red', 'Red', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 1324, 44, 'NDA', 1148427.165, 1851012.215, 41.74711093, -87.73170248, 70, 'ASHBURN', 13, 8, '(41.74711093, -87.73170248)')]

In [69]:
# List the type of schools along with the average safety score for each type.
%sql SELECT "Elementary, Middle, or High School" AS School_Type, AVG(SAFETY_SCORE) AS Average_Safety_Score FROM PublicSchools GROUP BY "Elementary, Middle, or High School";

 * sqlite:///FinalDB.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [72]:
#NO problem 7
%sql SELECT * from CensusData  LIMIT 3;

 * sqlite:///FinalDB.db
Done.


[(1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0),
 (2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0),
 (3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0)]

In [76]:
# List 5 community areas with highest % of households below poverty line
%sql SELECT COMMUNITY_AREA_NAME, PERCENT_HOUSEHOLDS_BELOW_POVERTY from CensusData ORDER BY PERCENT_HOUSEHOLDS_BELOW_POVERTY DESC LIMIT 5;

 * sqlite:///FinalDB.db
Done.


[('Riverdale', 56.5),
 ('Fuller Park', 51.2),
 ('Englewood', 46.6),
 ('North Lawndale', 43.1),
 ('East Garfield Park', 42.4)]

### Problem 8

##### Which community area is most crime prone? Display the coumminty area number only.


In [80]:
%sql SELECT COMMUNITY_AREA_NUMBER, COUNT(*) AS Crime_Count \
FROM CrimeData \
GROUP BY COMMUNITY_AREA_NUMBER \
ORDER BY Crime_Count DESC \
LIMIT 1;

 * sqlite:///FinalDB.db
Done.


[(25.0, 43)]


Double-click **here** for a hint

<!--
Query for the 'community area number' that has most number of incidents
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [83]:
%sql SELECT community_area_name \
FROM CensusData \
WHERE hardship_index = ( \
    SELECT MAX(hardship_index) \
    FROM CensusData \
);

 * sqlite:///FinalDB.db
Done.


[('Riverdale',)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [86]:
%sql SELECT community_area_name \
FROM CensusData \
WHERE community_area_number = ( \
    SELECT COMMUNITY_AREA_NUMBER \
    FROM CrimeData \
    GROUP BY COMMUNITY_AREA_NUMBER \
    ORDER BY COUNT(*) DESC \
    LIMIT 1 \
);


# Another way: 
# %sql SELECT community_area_name FROM CensusData WHERE COMMUNITY_AREA_NUMBER = 25.0;

 * sqlite:///FinalDB.db
Done.


[('Austin',)]

## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>
<h4>Abhishek Gagneja</h4>
<!--
## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
|2023-10-18  | 2.6     | Abhishek Gagneja  | Modified instruction set |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |
-->
## <h3 align="center"> © IBM Corporation 2023. All rights reserved. <h3/>
